db/access.php

```diff
@@ -24,10 +24,12 @@
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 
+use qtype_questionpy\constants;
+
 defined('MOODLE_INTERNAL') || die();
 
 $capabilities = [
-    'qtype/questionpy:uploadpackages' => [
+    constants::ROLE_UPLOAD => [
```

Comment:
Ist es problematisch, hier von einer anderen Datei abzuhängen? Ich habe dazu nichts gefunden.

Reply:
Das sollte unproblematisch sein, da es nur eine Konstante ist. [Bei der upgrade.php muss man etwas aufpassen](https://moodledev.io/docs/4.5/guides/upgrade#dbupgradephp). Meine Befürchtung wäre höchstens, dass der Autoloader während eines Upgrades (und in dem Zuge wird ja auch die access.php eingelesen) noch nicht funktioniert, aber in der Doku steht ja, dass Plugins Core-Funktionen benutzen dürfen, d.h. der Autoloader muss funktionieren.